macaroonbakery/tests/test_client.py

```diff
@@ -0,0 +1,376 @@
+            thread.start()
+            jar = requests.cookies.RequestsCookieJar()
+            with HTTMock(discharge):
+                with self.assertRaises(
+                        macaroonbakery.ThirdPartyCaveatCheckFailed):
+                    requests.get(
+                        url='http://' + httpd.server_address[0] + ':' +
+                            str(httpd.server_address[1]),
+                        cookies=jar,
+                        auth=httpbakery.BakeryAuth(cookies=jar))
+        finally:
+            httpd.shutdown()
+
+    def test_discharge_with_interaction_required_error(self):
+        class _DischargerLocator(macaroonbakery.ThirdPartyLocator):
+            def __init__(self):
+                self.key = macaroonbakery.generate_key()
+
+            def third_party_info(self, loc):
+                if loc == 'http://1.2.3.4':
+                    return macaroonbakery.ThirdPartyInfo(
+                        public_key=self.key.public_key,
+                        version=macaroonbakery.LATEST_BAKERY_VERSION,
+                    )
+        d = _DischargerLocator()
+        b = new_bakery('loc', d, None)
+
+        @urlmatch(path='.*/discharge')
+        def discharge(url, request):
+            return {
+                'status_code': 401,
+                'content': {
+                    'Code': httpbakery.ERR_INTERACTION_REQUIRED,
+                    'Message': 'interaction required',
+                    'Info': {
+                        'WaitURL': 'http://0.1.2.3/',
+                        'VisitURL': 'http://0.1.2.3/',
+                    },
+                }
+            }
+
+        def handler(*args):
+            GetHandler(b, 'http://1.2.3.4', None, None, None, *args)
+
+        try:
+            httpd = HTTPServer(('', 0), handler)
+            thread = threading.Thread(target=httpd.serve_forever)
+            thread.start()
+            jar = requests.cookies.RequestsCookieJar()
+
+            class MyInteractor(httpbakery.LegacyInteractor):
+                def legacy_interact(self, ctx, location, visit_url):
+                    raise httpbakery.InteractionError('cannot visit')
+
+                def interact(self, ctx, location, interaction_required_err):
+                    pass
+
+                def kind(self):
+                    return httpbakery.WEB_BROWSER_INTERACTION_KIND
+
+            with HTTMock(discharge):
+                with self.assertRaises(httpbakery.InteractionError):
+                    requests.get(
+                        'http://' + httpd.server_address[0] + ':' + str(
+                            httpd.server_address[1]),
+                        cookies=jar,
+                        auth=httpbakery.BakeryAuth(
+                            cookies=jar,
+                            interaction_methods=[MyInteractor()]))
+        finally:
+            httpd.shutdown()
+
+
+class GetHandler(BaseHTTPRequestHandler):
```

Comment:
Please document this class, even if briefly (yeah I know it's just for testing, but it helps).

Reply:
done